kahuna/public/js/services/api/leases.js

```diff
@@ -110,20 +128,48 @@ leaseService.factory('leaseService', [
       apiPoll(() => {
         return untilLeasesChange(images);
       }).then(results => {
-        $rootScope.$emit('images-updated', results.map(({image})=>image));
-        $rootScope.$emit('leases-updated');
-
-        return results.map(({leases}) => leases);
+          console.log(results);
+          return results.map(({ image, leases }) => {
+              emitLeaseUpdate(leases);
+              console.log(images, leases);
+                emitImageUpdate(image);
+          });
       });
     }
 
+    const imageUpdates$ = new Subject();
+        imageUpdates$.bufferWithTime(1000).subscribe((images) => {
```

Comment:
Oh cool, so we create a buffer, pump updates into it and issue updates periodically – basically a debounce?

Reply:
Yeah, basically, but we want to keep the id's about when we send the message.